### ISSUE TITLE
UI Interaction - Improving from ASM

### ISSUE THREAD

**Issue by shaunanoordin** [zooniverse/scribes-of-the-cairo-geniza#20]
## Discussion
@wgranger @beckyrother a few thoughts:

1. Do we really need an explicit Annotate Mode vs Navigation Mode?

Code-wise, we can detect the difference between a Click action and a Drag action. (Click = start annotating, Drag = start panning) The only question is if it's intuitive to users or not.

2. Do we want to reintroduce mouse scroll for zooming?

IIRC the main reason we removed this was because the zoom was too sensitive on Mac touchpads and/or other devices? i.e. an issue with implementation, not the concept itself.

3. We need to build an on-screen reminder/helper prompt.

Remember in ASM, when we're in Annotate/Transcribe mode and we click on the first dot - a popup appears? Prompting users is a good thing, but our implementation actually interrupts user actions, which I feel can be improved on.

For example, I'm thinking:
- User moves her cursor over the Subject. A tiny reminder pops up _unobtrusively in the corner of the screen_ and says "Click & drag to navigate, Click to start annotating."
- User clicks, placing the first dot. The tiny reminder now says "Annotating. Click again to place the endpoint, drawing a line underneath the text. Press ESC or click this (button) to cancel."

I'm thinking we need to build something like a cross between a Tooltip and a [Toast component](http://grommet.io/docs/toast/), except even less prominent. If this has already been foreseen by our design, hit me with a sad panda for not paying attention.

4. Actually, what kind of messaging systems have we built in, anyway?

Right now, we have one thing created that works really well: showing that a Subject is 1. Loading or 2. failed to load. 👍 (We should also reimplement that Project is Loading thing, come to think of it.)

Building on this kind of good UI design, how do we message...

- Your Classification is being sent, please hang on...
  - OK, success! Here's your next Subject.
  - Crap, that effect up, but don't worry, we saved it locally so you can retry later. (Assuming we have the queue future again.)
- DERP UNSPECIFIED ERROR pls contact our dev team

...? I need to think of what else I'm missing out on.

Basically what I'm saying is, "maybe we want a user-friendly equivalent to console.info() and alert() so they know what's happening at any given time."

**Comment by shaunanoordin** [comment]
Hahahaha, I just remembered - I know the perfect example of the kind of "Reminder Prompt" I'm thinking of, something to remind new users of the status/actions that are available to them at any given point in time, yet is also unobtrusive:

![mario-odyssey-cascade-kingdom-chain-chomp-open-retro-screenshot-2017-10-25-09-52-48](https://user-images.githubusercontent.com/13952701/35698892-52d50d08-0786-11e8-8f92-69e90f1ebba2.png)
_[Image source](http://www.usgamer.net/articles/18-12-2017-super-mario-odyssey-collectibles-guide-all-power-moon-locations-unlock-secrets-in-each-kingdom/super-mario-odyssey-cascade-kingdom-power-moon-locations-all-secrets)_

So in Super Mario Odyssey, Mario can possess different creatures, and when he does so, his control scheme changes. The unobtrusive reminder prompt at the bottom of the screen always ensures that the player isn't lost in figuring out what they can do next.

**Comment by beckyrother** [comment]
Hey @shaunanoordin! Thanks for putting this together. Some comments:

> Do we really need an explicit Annotate Mode vs Navigation Mode?
> Code-wise, we can detect the difference between a Click action and a Drag action. (Click = start annotating, Drag = start panning) The only question is if it's intuitive to users or not.

I think it's important that we explicitly tell users the action they're going to get when they click on the image. While it makes sense in the start annotating sense, it may not be obvious or intuitive to just be able to drag pan. Thinking about design software and the way both illustrator/photoshop and sketch handle pan actions, holding `spacebar` changes the pointer to a hand icon to indicate that click+drag = pan. Is this something we can do? 

I know `spacebar` on a browser will scroll down, but maybe it's possible to override that action here? Thoughts?

> Do we want to reintroduce mouse scroll for zooming?
> IIRC the main reason we removed this was because the zoom was too sensitive on Mac touchpads and/or other devices? i.e. an issue with implementation, not the concept itself.
Indeed, we removed it because throughout the zooniverse classifier, the zoom is extremely sensitive and turned out to be more of an encumbrance than a tool. If it's possible to tweak the sensitivity I'm happy to reconsider, _especially_ since this interface is full-screen by default and there _shouldn't_ be anything under the image to need to scroll to. Correct me if I'm wrong there though; if there is anything below the image to scroll to, we should _not_ enable the scroll to zoom in this case. 

> We need to build an on-screen reminder/helper prompt.
> Remember in ASM, when we're in Annotate/Transcribe mode and we click on the first dot - a popup appears? Prompting users is a good thing, but our implementation actually interrupts user actions, which I feel can be improved on.
> Basically what I'm saying is, "maybe we want a user-friendly equivalent to console.info() and alert() so they know what's happening at any given time."
Great point! I love the Mario example. What's really nice about Super Mario is that it gives the user maybe half a second to do the action on their own, and if they don't do anything, _then_ that helper text appears. So it doesn't talk down to the user, but quietly helps them out just in case they're feeling lost. 

Adding to my list :)

Thanks again Shaun! So glad you're thinking about these things.

**Comment by beckyrother** [comment]
![switch_supermarioodyssey_screen_01](https://user-images.githubusercontent.com/1392130/35754512-539baef6-0829-11e8-84cc-999ea7535f56.jpg)
such a great game
